### PR TITLE
[Feature] Enable response type selection for oauth2 implicit mode

### PR DESCRIPTION
**Pull request by emdfonseca**
This PR adds support for OpenID Connect by allowing the configuration of the OAuth 2 (implicit grant) response type. In the advanced options a new dropdown was added requesting the user to select whether he wants to receive the id_token, the access_token or both. By default the access_token is chosen (oauth 2 default). However, the user can also select to receive only the id_token or both the id_token and the access_token to be OpenID Connect compliant. 

This PR closes an already closed issue (#501). The issue was previously closed due to inactivity but the solution is not yet available. This PR should reopen the issue for further discussions and, hopefully, help improving insomnia with this new feature.

### CODE REVIEW

File: packages/insomnia-app/app/network/o-auth-2/grant-implicit.js [hunk 1]
```diff
@@ -9,30 +9,33 @@ export default async function (
   clientId: string,
   redirectUri: string = '',
   scope: string = '',
-  state: string = ''
+  state: string = '',
+  responseType: string = ''
```
Comment (gschier) [inline packages/insomnia-app/app/network/o-auth-2/grant-implicit.js]: [`response_type`](https://tools.ietf.org/html/rfc6749#section-3.1.1) is a required OAuth 2.0 parameter so this argument default should probably be changed from `''` to `c.RESPONSE_TYPE_TOKEN`;

Comment (gschier) [reply packages/insomnia-app/app/network/o-auth-2/grant-implicit.js]: And line 15 can be reverted.

File: packages/insomnia-app/app/network/o-auth-2/misc.js [hunk 2]
```diff
@@ -45,24 +63,24 @@ export function authorizeUserInWindow (url, urlRegex = /.*/) {
...
-      } else if (currentUrl === url) {
-        // It's the first one, so it's not a redirect
-        console.log(`[oauth2] Loaded "${currentUrl}"`);
-      } else {
-        console.log(`[oauth2] Ignoring URL "${currentUrl}". Didn't match ${urlRegex.toString()}`);
-      }
+      _parseUrl(currentUrl);
+    });
+
+    child.webContents.on('did-fail-load', (e, errorCode, errorDescription, url) => {
```
Comment (gschier) [inline packages/insomnia-app/app/network/o-auth-2/misc.js]: Nice 👍

**Comment by emdfonseca** [comment]
Hi @gschier, thanks for the feedback! I made the changes as requested.

**Comment by emdfonseca** [comment]
Hi @gschier, any additional changes needed in order to get this PR merged?

**Comment by gschier** [comment]
Sorry, haven't had time to look at it again. It's probably good enough to merge at this point. I'll try to get it merge this week.

**Pull request merged**